src/utilities/__tests__/separateOperations-test.js

```diff
@@ -0,0 +1,124 @@
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import { describe, it } from 'mocha';
+import { expect } from 'chai';
+import { separateOperations } from '../separateOperations';
+import { parse, print } from '../../language';
+
+
+describe('separateOperations', () => {
+
+  it('separates one AST into multiple, maintaining document order', () => {
```

Comment:
Good idea. Circular dependency is a validation error, but we certainly don't want our tools to break when operating on invalid queries either.